src/fips.c

```diff
@@ -310,17 +314,36 @@ static EVP_PKEY *get_pkey(void)
 	return pkey;
 }
 
+static void* mmap_file(const char *path, struct stat *statbuf)
+{
+	int fd;
+	void *fp = NULL;
+
+	fd = open(path, O_RDONLY);
+	if (fd < 0)
+		goto end;
+
+	if (fstat(fd, statbuf) < 0)
+		goto end_close;
+
+	fp = mmap(0, statbuf->st_size, PROT_READ, MAP_SHARED, fd, 0);
+	if (fp == MAP_FAILED)
+		fp = NULL;
+end_close:
```

Comment:
isn't this always closing the file ? I guess a `goto end;` is missing in the good case-

Reply:
In general, if a file is accessed via mmap, the fd can be closed as soon as the file is successfully mmap-ed in address space of the process. This is the common file-mmap processing in unix/linux projects.

So: We're always closing the file and do not need a `goto end;` for the good case.

Reply:
Ah sure, my fault. I thought you are returning the fd, but you are returning the mapped pointer.
All is good. Sorry for the noise.

Reply:
np, I also thought so during the review before sending the PR... :) Maybe I should rename fp to fdata in this function as well.

Reply:
> Maybe I should rename fp to fdata ...

that would help.

Reply:
ok, I will change it.